### PR TITLE
Adding helpers to Config, PubSubMessage, Metadata. Coverage improved

### PR DESCRIPTION
**Pull request by akshaisarma**

### CODE REVIEW

File: src/main/java/com/yahoo/bullet/pubsub/Metadata.java [hunk 1]
```diff
@@ -5,21 +5,56 @@
...
+        COMPLETE,
+        FAIL
     }
+
+    private static final long serialVersionUID = 4234800234857923112L;
+
+    @Getter @Setter
     private Signal signal;
-    private Serializable content;
+    private Object content;
```
Comment (hbxie) [inline src/main/java/com/yahoo/bullet/pubsub/Metadata.java]: Why did you change from Serializable to Object and do the type cast after?

Comment (akshaisarma) [reply src/main/java/com/yahoo/bullet/pubsub/Metadata.java]: If the type is Serializable, GSON cannot create an instance of it. So I kept the interface the same (allowing only Serializable objects in or out) but store it as an object. The casting is only done on the getter.  GSON is used in PubSubMessage#fromJSON. See a better way?

Comment (hbxie) [reply src/main/java/com/yahoo/bullet/pubsub/Metadata.java]: Then add a comment here?

Comment (akshaisarma) [reply src/main/java/com/yahoo/bullet/pubsub/Metadata.java]: Will do.
